src/client/WebhookClient.ts

```diff
@@ -18,7 +18,7 @@ export class WebhookClient extends BaseClient {
 	 * @param token The webhook token
 	 */
 	public async fetch(id: string, token?: string): Promise<Webhook> {
-		const webhook = await Webhook.fetch(this, id, token);
+		const webhook = await (token ? Webhook.fetch(this, id, token) : Webhook.fetch(this, id));
```

Comment:
I made Webhook.fetch overloaded so it'd show different documentation for the version with and without token.

But apparently TS considers them 2 different things instead of getting the result of all overloads joined, despite of their returns being similar, so feels like a bug in their side.

I can revert the overload documentation so we don't make branches here, but I fear lower docs quality.

Reply:
I don't feel like we should be "fixing typescript" on our own. We should write it right, and typescript will fix itself over time.